chore: Filter SDK messages in logger

limbonaut opened the pull request:
With new logger implementation, Sentry SDK messages end up as breadcrumbs in Sentry. This PR adds filtering to exclude such messages.

Resolves #231.

<img width="726" alt="Screenshot 2025-07-04 at 13 16 00" src="https://github.com/user-attachments/assets/c5505fae-4d16-4825-88d6-e10c2651ea72" />

jpnurmi commented:
Could there be an internal API for printing the debug logs that should not be sent to Sentry, to avoid the cost of processing every log line with a regexp to filter out lines that shouldn't have been there in the first place?

limbonaut commented:
@jpnurmi Sadly this is the cost of usability. If we intend for those messages to end up in the Godot output (and the log file):

<img width="600" alt="screen" src="https://github.com/user-attachments/assets/c711cd37-53a7-4747-bfe7-83c5d4771ebd" />

limbonaut commented:
I've updated the PR. Here are the changes:
- Reformat sentry output for easier filtering.
- Remove regex usage for performance reasons.
- Use filtering based on prefix matching.
- Additionally, remove attachment warning message filtering.

Filtering a message now takes 0-1 microsecond instead of 11-15, not counting attachment warning gains.

I also tested how much headroom this approach has, and added a couple more prefixes to the filter array, and it's still 1 usec.

limbonaut commented:
@sentry review

limbonaut commented on src/sentry_logger.h:
```diff
@@ -44,6 +44,9 @@ class SentryLogger : public Logger {
 	// Number of events captured during this frame.
 	int frame_events = 0;
 
+	// Filter: Remove messages from breadcrumbs which start with any of these prefixes.
+	std::vector<String> filter_by_prefix;
```
I've tested it first, PackedStringArray or godot::Vector<String> is a little slower.
